src/commands/Quotes/addQuote.ts

```diff
@@ -0,0 +1,142 @@
+      quoteAuthor = await args.pickResult('string');
+
+      if (!quote.success || !quoteAuthor.success) return invalidSyntaxReply(message, settingsData);
+      quote = quote.value;
+    }
+
+    if (quote.length > EMBED_FIELD_MAX_CHAR_LENGTH) {
+      const embed = new BediEmbed()
+          .setColor(colors.ERROR)
+          .setTitle('Add Quote Reply')
+          .setDescription('Quote is too long! Please submit a quote that is 1024 characters or fewer.');
+      return message.reply({embeds: [embed]});
+    }
+
+    const embed = new BediEmbed()
+        .setTitle(`Add Quote Reply - Approvals: 0/${settingsData.quoteApprovalsRequired}`)
+        .setDescription(`Quote: ${surroundStringWithBackTick(quote)}
+        Author: ${surroundStringWithBackTick(quoteAuthor.value as string)}
+        Submitted By: ${author}
+        Approved By:`);
+
+    const row = new MessageActionRow()
+        .addComponents(
+            new MessageButton()
+                .setCustomId('QuoteApprove')
+                .setLabel('Approve')
+                .setStyle('SUCCESS'),
+        );
+
+    const response = await message.reply({
+      embeds: [embed],
+      components: [row],
+    });
+  }
+
+  async onLoad() {
+    client.on('interactionCreate', async interaction => {
+      if (!interaction.isButton() || !(interaction.customId === 'QuoteApprove')) return;
+      await interaction.deferUpdate();
+
+      const message = interaction.message;
+      if (!(message instanceof Message)) return;
+
+      let description = message.embeds[0].description;
+      if (description?.includes(interaction.user.toString())) return;
+
+      const title = message.embeds[0].title;
+      let numApprovals = parseInt(title?.substring('Add Quote Reply - Approvals: '.length, title.indexOf('/')) as string, 10);
+      numApprovals++;
+      description += ` ${interaction.user}`;
+
+      const settingsData = await getSettings(interaction.guildId as string);
+
+      if (numApprovals < settingsData.quoteApprovalsRequired) {
+        const embed = new BediEmbed()
+            .setTitle(`Add Quote Reply - Approvals: ${numApprovals}/${settingsData.quoteApprovalsRequired}`)
+            .setDescription(description as string);
+
+        await message.edit({embeds: [embed]});
+      } else {
+        const embed = new BediEmbed()
+            .setTitle('Add Quote Reply - Approved')
+            .setDescription(description as string);
+
+        const quote = description?.substring('Quote: `'.length, description?.indexOf('`', 'Quote: `'.length + 1));
+        const authorIndex = (description?.indexOf('Author: `') as number) + 'Author: `'.length;
+        const author = description?.substring(authorIndex, description?.indexOf('`', authorIndex + 1));
+
+        await addQuote(interaction.guildId as string, quote as string, author as string);
+
+        await message.edit({
+          embeds: [embed],
+          components: [],
+        });
```

Comment:
What prevents the else statement from executing if the number of approvals goes over 4?

Reply:
The else branch removes the button from the message entirely.